test: improve coverage for TraceDiffGraph directory

vishvamsinh28 opened the pull request:
This PR improves test coverage for renderNode, and traceDiffGraphUtils by adding tests that cover previously untested branches. These additions help achieve full coverage.

Before
<img width="1440" alt="Screenshot 2025-05-21 at 11 09 59 PM" src="https://github.com/user-attachments/assets/17ebbe4e-caa4-40c0-8150-dc3c3338f702" />

After
<img width="1439" alt="Screenshot 2025-05-21 at 11 08 07 PM" src="https://github.com/user-attachments/assets/6454e471-2b58-40fe-a03a-4ec8ea88b9cb" />

yurishkuro commented on packages/jaeger-ui/src/components/TraceDiff/TraceDiffGraph/TraceDiffGraph.test.js:
```diff
@@ -168,4 +168,27 @@ describe('TraceDiffGraph', () => {
+      state: fetchedState.DONE,
+    };
+
+    wrapper = shallow(<TraceDiffGraph a={baseTrace} b={baseTrace} uiFind="GET" />);
```
if you are adding new tests can you write them in RTL style, without using enzyme's `shallow()`?

vishvamsinh28 replied:
yes

vishvamsinh28 replied:
TraceDiffGraph.test.js uses Enzyme's shallow; should i rewrite all the tests in RTL style?

vishvamsinh28 commented:
@yurishkuro Migrated TraceDiffGraph.test.js file from Enzyme to RTL

1. Replaced Enzyme's shallow rendering with RTL's render and screen utilities
2. Removed wrapper.setProps and used rerender instead for prop updates
3. Dropped instance-based test (layoutManager.stopAndRelease) since RTL does not support direct instance access and the cleanup logic is considered an internal implementation detail
5. Mocked UiFindInput, ErrorMessage, and LoadingIndicator for simpler and more focused assertions
6. Preserved all original behavior and logic tests including error, loading, uiFind, and graph rendering
7. Snapshot testing retained using RTL's container.toMatchSnapshot
8. Ran `npm test -- -u` to update snapshots and remove obsolete Enzyme-based snapshot

vishvamsinh28 commented:
@yurishkuro Migrated test files from Enzyme to RTL

- Changes in TraceDiffGraph.test.js :

1. Replaced Enzyme's shallow rendering with RTL's render and screen utilities
2. Removed wrapper.setProps and used rerender instead for prop updates
3. Dropped instance-based test (layoutManager.stopAndRelease) since RTL does not support direct instance access and the cleanup logic is considered an internal implementation detail
5. Mocked UiFindInput, ErrorMessage, and LoadingIndicator for simpler and more focused assertions
6. Preserved all original behavior and logic tests including error, loading, uiFind, and graph rendering
7. Snapshot testing retained using RTL's container.toMatchSnapshot
8. Ran `npm test -- -u` to update snapshots and remove obsolete Enzyme-based snapshot

- Changes in renderNode.test.js :

1. Replaced Enzyme's shallow rendering with RTL's render and cleanup utilities
2. Removed wrapper.setProps in favor of re-rendering component with updated props
3. Snapshot testing retained using RTL's container.toMatchSnapshot()
4. Test descriptions reworded slightly for clarity and RTL style, but all original behaviors preserved
5. No component logic was changed, only the test implementation was modernized

yurishkuro commented on packages/jaeger-ui/src/components/TraceDiff/TraceDiffGraph/renderNode.test.js:
```diff
@@ -13,63 +13,68 @@
-      expect(wrapper).toMatchSnapshot();
+    it('renders correctly when a < b', () => {
+      const { container } = render(<DiffNode {...baseProps} a={defaultCount / 2} />);
+      expect(container.querySelector('.is-more')).toBeInTheDocument();
```
how does this test that `a < b`? The `.is-more` style could be on either side, no?

yurishkuro replied:
relying on `toContain('+')` seems unstable. Can't we use more robust lookup, but test id or some class name that's different between left and right?

vishvamsinh28 replied:
I’ve added data-testid attributes to the metric cells and updated all related tests to use screen.getByTestId instead of relying on textContent. Let me know if you'd like any other changes! 🙌
